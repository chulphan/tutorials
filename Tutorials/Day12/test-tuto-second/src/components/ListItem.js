import React from 'react';

function ListItem(props) {
    const { item } = props;
    return <li className="item">{item}</li>
}

export default ListItem;