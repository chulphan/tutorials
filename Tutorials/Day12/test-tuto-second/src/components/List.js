import React from 'react';

import ListItem from './ListItem';

function List(props) {
    const { items } = props;
    if (!items.length) {
        return <span className="empty-message">No items in list</span>;
    }

    return (
        <ul className="list-items">
            { items.map(item => <ListItem key={item} item={item}/> )}
        </ul>
    );
}

List.defaultProps = {
    items: []
};

export default List;