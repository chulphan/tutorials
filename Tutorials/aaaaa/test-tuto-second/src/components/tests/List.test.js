import React from 'react';
import { shallow, mount, render } from 'enzyme';

import '../../setupEzm';

import List from '../List';
import ListItem from '../ListItem';

describe('List tests', () => {

    it('renders list-items', () => {
        const items = ['one', 'two', 'three'];
        //const wrapper = shallow(<List items={items} />);

        const wrapper = render(<List items={items} />);

        wrapper.addClass('foo');

        expect(wrapper.find('.list-items')).toBeDefined();
        expect(wrapper.find('.item')).toHaveLength(items.length);
    });

    it('renders a list item', () => {
        const items = ['Thor', 'Loki'];
        const wrapper = mount(<List items={items} />);

        expect(wrapper.contains(<li key='Thor' className="item">Thor</li>)).toBeTruthy();
    });

    it('renders correct text in item', () => {
        const items = ['John', 'James', 'Luke'];
        const wrapper = mount(<List items={items} />);

        expect(wrapper.find('.item').get(0).props.children).toEqual('John');
    });

    it('renders no item', () => {
        const items = [];
        const wrapper = mount(<List items={items} />);

        expect(wrapper.contains(<span className="empty-message">No items in list</span>)).toBeTruthy();
    });

    it('matches snapshot', () => {

        const items = ['one', 'two', 'three'];

        const wrapper = render(<List items={items} />);

        expect(wrapper).toMatchSnapshot();
    });
})