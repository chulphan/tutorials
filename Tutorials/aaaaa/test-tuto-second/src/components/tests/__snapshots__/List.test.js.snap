// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List tests matches snapshot 1`] = `
<ul
  class="list-items"
>
  <li
    class="item"
  >
    one
  </li>
  <li
    class="item"
  >
    two
  </li>
  <li
    class="item"
  >
    three
  </li>
</ul>
`;

exports[`matches snapshot 1`] = `
<ul
  class="list-items"
>
  <li
    class="item"
  >
    one
  </li>
  <li
    class="item"
  >
    two
  </li>
  <li
    class="item"
  >
    three
  </li>
</ul>
`;
